### PR TITLE
[SPARK-9002][CORE] KryoSerializer initialization does not include 'Array[Int]'

### PR DESCRIPTION
**Pull request by dbolshak**
[SPARK-9002][CORE] KryoSerializer initialization does not include 'Array[Int]'

## What changes were proposed in this pull request?

Array[Int] has been registered in KryoSerializer.
The following file has been changed
core/src/main/scala/org/apache/spark/serializer/KryoSerializer.scala

## How was this patch tested?

First, the issue was reproduced by new unit test.
Then, the issue was fixed to pass the failed test.

### CODE REVIEW

**Comment by srowen** [comment]
Yeah I had also wondered this, why things like arrays of doubles, floats, chars, Strings, etc aren't included. I don't see why not.

**Comment by srowen** [comment]
Jenkins add to whitelist

File: core/src/test/scala/org/apache/spark/serializer/KryoSerializerSuite.scala [hunk 1]
```diff
@@ -339,6 +339,16 @@ class KryoSerializerSuite extends SparkFunSuite with SharedSparkContext {
...
+  test("registration of Array[Int]") {
+    val conf = new SparkConf(false)
+    conf.set("spark.kryo.registrationRequired", "true")
+
+    val ser = new KryoSerializer(conf).newInstance()
+
+    val t = Array(0, 1, 2)
+    assert(ser.deserialize[Array[Int]](ser.serialize(t)) === t)
+  }
+
```
Comment (BenFradet) [inline core/src/test/scala/org/apache/spark/serializer/KryoSerializerSuite.scala]: Why not add `registrationRequired` to the tests above (l78)?

**Comment by dbolshak** [comment]
@srowen, @BenFradet Please take a look.

File: core/src/test/scala/org/apache/spark/serializer/KryoSerializerSuite.scala [hunk 2]
```diff
@@ -76,6 +76,9 @@ class KryoSerializerSuite extends SparkFunSuite with SharedSparkContext {
   }
 
   test("basic types") {
+    val conf = new SparkConf(false)
```
Comment (srowen) [inline core/src/test/scala/org/apache/spark/serializer/KryoSerializerSuite.scala]: You could do this to the next test too, and then, I think you'd have to register Tuple2, but that sounds like a good idea.

And then the next for Scala Map, Seq, List. Again, probably a good idea, as in the JIRA. I actually don't know why this isn't done already.

Comment (dbolshak) [inline core/src/test/scala/org/apache/spark/serializer/KryoSerializerSuite.scala]: Unit test have been updated, nothing new was found.

Tuples are registered starting from line 149 in KryoSerializer.scala.

**Review by BenFradet — APPROVED** [review]
LGTM :+1:

**Comment by dbolshak** [comment]
@srowen Is it all right if I ask you to merge the PR?

**Comment by srowen** [comment]
I would not ping within a few days. We leave changes open for comment for a while if it's not a rush.

**Comment by dbolshak** [comment]
Thank you.

**Comment by srowen** [comment]
I wonder if @davies @zsxwing @squito know of any reason this wouldn't be a good idea? seems like an obvious thing to do but I wonder what I might be missing. Maybe it's just one of those things that was added to on a case-by-case basis as needed

**Comment by srowen** [comment]
Merged to master

**Comment by srowen** [comment]
@dbolshak let me know your JIRA handle and I'll credit you at https://issues.apache.org/jira/browse/SPARK-9002

**Pull request closed without merging**